File: 189-旋转数组/189-1.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int s = nums.size();
        k = k%s;
        reverse(nums.begin(), nums.begin() + s - k);
        reverse(nums.begin()+s-k, nums.begin() + s);
        reverse(nums.begin(), nums.begin() + s);
    }
};

File: 189-旋转数组/189-1.java
class Solution {
  public void rotate(int[] nums, int k) {
      int n = nums.length;
      k %= n;
      int[] res = new int[n];
      int j = 0;
      for(int i = n-k;i<n;i++){
          res[j++] = nums[i];
      }
      for(int i = 0;i<n-k;i++){
          res[j++] = nums[i];
      }
      for(int i = 0; i<n;i++){
          nums[i] = res[i];
      }
  }
}